### PR TITLE
Remove automatic reboots.

### PR DESCRIPTION
**Pull request by denravonska**
This removes the Windows specific automatic reboots which might be what some people are seeing when they come back to a shut down wallet.

The logic for CheckFutileSync was also changed since it did not make sense once the reboot was removed. Needs double checking.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -4780,19 +4730,32 @@ bool ProcessBlock(CNode* pfrom, CBlock* pblock, bool generated_by_me)
...
+                // We reset these counters every time a block is accepted successfully in AcceptBlock().
+                // Note: This code will never actually be exercised unless the wallet stays out of sync for a very long time - approx. 24 hours - the wallet normally recovers on its own without this code.
+                // I'm leaving this in for people who may be on vacation for a long time - it may keep an external node running when everything else fails.
+                if (TimerMain("CheckForFutileSync", 25))
                 {
-                    mapAlreadyAskedFor.clear();
-                    printf("\r\nClearing mapAlreadyAskedFor.\r\n");
-                    AskForOutstandingBlocks(uint256(0));
-                    CheckForFutileSync();
+                    mapOrphanBlocks.clear();
```
Comment (tomasbrod) [inline src/main.cpp]: why clear this map?

**Review by tomasbrod — APPROVED** [review]

File: src/main.cpp (hunk 1 above)
Comment (denravonska) [reply src/main.cpp]: Tbh I just moved the contents of `CheckForFutileSync` down here. I don't know if this needs be be refactored further.

Comment (tomasbrod) [reply src/main.cpp]: It looks shady to me. But unrelated to this change. I Still approve.

Comment (pwnorbitals) [reply src/main.cpp]: The map clearing will leak memory, switch to `clearOrphanBlocks()` when https://github.com/gridcoin/Gridcoin-Research/pull/609 is merged ;-)  (or will it be done automatically ? )

**Comment by tomasbrod** [comment]
Why this was not applied for the hotfix?

**Comment by denravonska** [comment]
@tomasbrod It's not that tested yet. It can wait for the next leisure.

**Pull request merged**